peerconnection.go

```diff
@@ -1079,15 +1171,119 @@ func (pc *PeerConnection) drainSRTP() {
 				return
 			}
 
-			_, ssrc, err := srtpSession.AcceptStream()
+			r, ssrc, err := srtpSession.AcceptStream()
 			if err != nil {
 				pc.log.Warnf("Failed to accept RTP %v", err)
 				return
 			}
 
-			if !handleUndeclaredSSRC(ssrc) {
-				pc.log.Warnf("Incoming unhandled RTP ssrc(%d), OnTrack will not be fired", ssrc)
-			}
+			pc.mu.Lock()
+			pc.pendingReadStreamsSRTP[ssrc] = r
+			pc.mu.Unlock()
+
+			go func() {
+				// test first N (10) packets, if no mid and rid are found give up
+				c := 0
+
+				// read incoming packet until we can populate mid and rid from packet
+				// extensions (not all packet will provide such information) so continue reading until we'll find both
+				var mid, rid string
+				for {
+					b := make([]byte, receiveMTU)
+					i, err := r.Read(b)
+					if err != nil {
+						pc.log.Errorf("Failed to read RTP %v", err)
+						return
+					}
+
+					c++
+
+					if c >= 10 {
+						// no rid enabled receiver matches to found mid
+						if !handleUndeclaredSSRC(ssrc) {
+							pc.log.Warnf("Incoming unhandled RTP ssrc(%d), OnTrack will not be fired", ssrc)
+						}
+
+						return
+					}
+
+					rp := &rtp.Packet{}
+					if err := rp.Unmarshal(b[:i]); err != nil {
+						pc.log.Errorf("Failed to unmarshal RTP %v", err)
+						return
+					}
+
+					if !rp.Header.Extension {
+						continue
+					}
+
+					var sdesMidExtMap *sdp.ExtMap
+					var sdesStreamIDExtMap *sdp.ExtMap
+					for _, extMap := range pc.extMaps {
+						switch extMap.URI.String() {
+						case sdesMidURI.String():
+							sdesMidExtMap = extMap
+						case sdesRTPStreamIDURI.String():
+							sdesStreamIDExtMap = extMap
+						}
+					}
+
+					if payload := rp.GetExtension(uint8(sdesMidExtMap.Value)); payload != nil {
+						curMid := string(payload)
+						pc.log.Infof("curMid: %q", curMid)
```

Comment:
Arguably some of this is a bit noisy for info level

Reply:
Just a debugging refuse. Removed